src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -544,19 +545,44 @@ void regclass_Core(py::module m) {
            const py::object& model_stream,
            const std::string& device_name,
            const std::map<std::string, py::object>& properties) {
-            auto _properties = Common::utils::properties_to_any_map(properties);
+            const auto _properties = Common::utils::properties_to_any_map(properties);
             if (!(py::isinstance(model_stream, pybind11::module::import("io").attr("BytesIO")))) {
                 throw py::type_error("CompiledModel.import_model(model_stream) incompatible function argument: "
                                      "`model_stream` must be an io.BytesIO object but " +
                                      (std::string)(py::repr(model_stream)) + "` provided");
             }
             model_stream.attr("seek")(0);  // Always rewind stream!
-            std::stringstream _stream;
-            _stream << model_stream
-                           .attr("read")()  // alternative: model_stream.attr("get_value")()
-                           .cast<std::string>();
-            py::gil_scoped_release release;
-            return self.import_model(_stream, device_name, _properties);
+            const std::string filename = "model_stream.txt";
+            std::fstream _stream(filename, std::ios::out | std::ios::binary);
```

Comment:
1. do we need to remove this temporary file once we have performed model importing?
2. since std::fstream is slower than std::stringstream, should we use std::stringstream for cases < 2GB? I suppose we can detect it based on `model_stream` size

Reply:
> do we need to remove this temporary file once we have performed model importing?
I don't think we necessarily do, I just wanted to avoid leaving big temp files in the users' file system. If the file is not deleted and `import_model` is called again, the temp file will be overwritten anyway.

> since std::fstream is slower than std::stringstream, should we use std::stringstream for cases < 2GB? I suppose we can detect it based on model_stream size
Probably a good idea, but I'd suggest adding it to CVS-148126 - the `fstream` fix is for unblocking the NPU, but it has to be introduced to other places in the code as well. Since branching the implementation based on `model_stream` size would require additional testing, I'm in favor of unblocking the NPU and making improvements in the scope of the ticket I mentioned.

Reply:
> I don't think we necessarily do

If the model is 70B size, you will store temporary big files on users's FS, which is not good IMO.

I vote for cleaning up